### PR TITLE
fix: compile warnings

### PR DESCRIPTION
**Pull request by nnyyxxxx**
- Cleans up compile warnings
- Formats code
- Cleans up unused vars
- etc

Tested

### CODE REVIEW

File: include/hyprutils/os/Process.hpp [hunk 1]
```diff
@@ -25,7 +25,7 @@ namespace Hyprutils {
             const std::string& stdErr();
 
             // only populated when ran async
-            const pid_t pid();
+            const pid_t& pid();
```
Comment (vaxerski) [inline include/hyprutils/os/Process.hpp:28]: no api breaking

Comment (nnyyxxxx) [reply include/hyprutils/os/Process.hpp:28]: that doesnt break the api, its just returning a reference instead of a copy

Comment (nnyyxxxx) [reply include/hyprutils/os/Process.hpp:28]:
```cpp
const std::string& Hyprutils::OS::CProcess::stdOut() {
    return out;
}

const std::string& Hyprutils::OS::CProcess::stdErr() {
    return err;
}
```

These are doing the same thing, its just to keep consistency

Comment (vaxerski) [reply include/hyprutils/os/Process.hpp:28]: breaks ABI for sure

File: tests/string.cpp [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 using namespace Hyprutils::String;
 
-int main(int argc, char** argv, char** envp) {
```
Comment (vaxerski) [inline tests/string.cpp:7]: why remove my signature trademark `int argc char** argv char** envp`

**Review by vaxerski — COMMENTED** [review]
rest ok

File: tests/string.cpp (hunk 2 above)
Comment (nnyyxxxx) [reply tests/string.cpp:7]: they arent being used

Comment (vaxerski) [reply tests/string.cpp:7]: but it's a trademark of mine

Comment (nnyyxxxx) [reply tests/string.cpp:7]: ???

Comment (nnyyxxxx) [reply tests/string.cpp]: [a922578](https://github.com/hyprwm/hyprutils/pull/25/commits/a922578e9c6acca4f932d20fc39dd5d574980291) added back here

**Review by vaxerski — APPROVED** [review]
thanks ganyu

**Pull request merged**